### DISCUSSION TITLE
p-fileupload progressbar for each file

### DISCUSSION THREAD

**Discussion by mahraz-hasnat** [primefaces/community#2628] [PrimeNG] [answered]
Hi everyone,

I'm new to Angular and was wondering if there's a way to use p-fileUpload with a progress bar for each individual file. Any guidance would be greatly appreciated!

The backend is in .NET API and the storage is Azure Blob.

**Comment by qwadrox** [comment, marked as answer]
The component does not support tracking the progress of individual files. You might be able to achieve this by relying on the advanced templating, CSS, and [customUpload](https://primeng.org/fileupload#api.fileupload.props.customUpload). 

To give you some idea: 

```ts
import { Component } from '@angular/core';

@Component({
  selector: 'app-file-upload',
  templateUrl: './file-upload.component.html'
})
export class FileUploadComponent {

  // In the template iterate over the files and do something like <p-progressBar [value]="progress[file.name]" ></p-progressBar>

  progress: { [key: string]: number } = {};

  uploadHandler({files}) {
      files.forEach((file: any) => {
          // Upload the files and 
          // calculate the progress for each file individually
          this.progress[file.name] = Math.round((e.loaded * 100) / e.total);
    });
  }
}
```
Or you can create your own simplified version of upload component that matches your needs. Here is the original code you can gather some idea: https://github.com/primefaces/primeng/blob/1f6c5d95271e5b48aabdcd15bc040a1f3bea5adc/src/app/components/fileupload/fileupload.ts#L46

**Reply by mahraz-hasnat** [reply]
thanks
